sssom/util.py

```diff
@@ -1250,3 +1250,19 @@ def get_all_prefixes(msdf: MappingSetDataFrame) -> list:
         prefix_list = list(set(prefix_list))
 
     return prefix_list
+
+
+def augment_metadata(msdf: MappingSetDataFrame, meta: dict) -> MappingSetDataFrame:
+    """Augment metadata with parameters passed.
+
+    :param msdf: MappingSetDataFrame (MSDF) object.
+    :param meta: Dictionary that needs to be added/updated to the metadata of the MSDF.
+    :return: MSDF with updated metadata.
+    """
+    if msdf.metadata:
+        for k, v in meta.items():
+            if len(v) == 1:
+                msdf.metadata[k] = v[0]
+            else:
+                msdf.metadata[k] = list(v)
```

Comment:
The problem with this is that if have have to creators in the metadata and want to add a third one.. it wont work. Would it make sense to make the method strictly additive for multi valued metadata? Maybe not. Do you have an opinion on this? Maybe it is correct the way it is..

Reply:
Good catch! We will have to think about it. When is it additive and when is it replacement? Should there be a flag?

Reply:
So what I have implemented now is if a `slot` to be annotated is `multivalued` , add new values to the list. But what if the entire `slot` value needs to be changed? How does the user indicate that?

Reply:
What about a boolean flag `--replace-multivalued` which defaults to false, and if it is true, it will remove any existing values from a multivalued value?

Reply:
But what if the users want to retain some of the contents of the list and add new stuff? They'll have to type the ones they want again. If that's the case, the first solution would have worked and was a simple one too.

Reply:
For now I implemented `replace_multivalued`. If we can think of a cleaner way to do this, we can always go back and improve the code.

Reply:
I think this is fine. good one :)